pyiron_contrib/protocol/compound/finite_temperature_string.py

```diff
@@ -290,420 +354,86 @@ def get_output(self):
-        g.reflect_string.broadcast.previous_velocities = gp.reflect_atoms.output.velocities[-1]
-        g.reflect_string.broadcast.previous_forces = gp.reflect_atoms.output.forces[-1]
-
-        g.reflect_string.broadcast.positions = gp.verlet_positions.output.positions[-1]
-        g.reflect_string.broadcast.velocities = gp.verlet_velocities.output.velocities[-1]
-        g.reflect_string.broadcast.forces = gp.calc_static.output.forces[-1]
-
-        g.reflect_string.direct.all_centroid_positions = ip.all_centroid_positions
-        g.reflect_string.broadcast.centroid_positions = ip.all_centroid_positions
-
-        # reflect individual atoms which stray too far
-        g.reflect_atoms.input.n_children = ip.n_images
-        g.reflect_atoms.direct.cutoff_distance = ip.reflection_cutoff_distance
-        g.reflect_atoms.direct.cell = ip.structure.cell
-        g.reflect_atoms.direct.pbc = ip.structure.pbc
-
-        g.reflect_atoms.broadcast.default.previous_positions = ip.all_centroid_positions
-        g.reflect_atoms.broadcast.default.previous_velocities = gp.initial_velocities.output.velocities[-1]
-        g.reflect_atoms.direct.default.previous_forces = gp.initial_forces.output.zeros[-1]
-
-        g.reflect_atoms.broadcast.previous_positions = gp.reflect_atoms.output.positions[-1]
-        g.reflect_atoms.broadcast.previous_velocities = gp.reflect_atoms.output.velocities[-1]
-        g.reflect_atoms.broadcast.previous_forces = gp.reflect_atoms.output.forces[-1]
-
-        g.reflect_atoms.broadcast.positions = gp.reflect_string.output.positions[-1]
-        g.reflect_atoms.broadcast.velocities = gp.reflect_string.output.velocities[-1]
-        g.reflect_atoms.broadcast.forces = gp.reflect_string.output.velocities[-1]
-
-        g.reflect_atoms.broadcast.reference_positions = ip.all_centroid_positions
-
-        # check_thermalized
-        g.check_thermalized.input.target = gp.clock.output.n_counts[-1]
-        g.check_thermalized.input.threshold = ip.thermalization_steps
+            return self.graph.calc_static_centroids.archive.output.energy_pot.data[frame]
+
+    def plot_string(self, ax=None, frame=None, plot_kwargs=None):
+        if ax is None:
+            _, ax = plt.subplots()
+        if plot_kwargs is None:
+            plot_kwargs = {}
+        if 'marker' not in plot_kwargs.keys():
+            plot_kwargs = {'marker': 'o'}
+        energies = np.array(self._get_energies(frame=frame))
+        ax.plot(energies - energies[0], **plot_kwargs)
+        ax.set_ylabel("Energy")
+        ax.set_xlabel("Centroid")
+        ax.xaxis.set_major_locator(MaxNLocator(integer=True))
+        return ax
+
+    def get_forward_barrier(self, frame=None):
+        """
+        Get the energy barrier from the 0th image to the highest energy (saddle state).
+        """
+        energies = self._get_energies(frame=frame)
+        return np.amax(energies) - energies[0]
```

Comment:
I'm okay with either of them. Both of them take the same time to run because I guess they are the same thing under the hood? Personally, I am slightly more biased towards .max() for visual reasons, as is it shorter than np.amax(). -.-'